Functions/Mock.ps1

```diff
@@ -1560,7 +1560,7 @@ function Repair-ConflictingParameters {
     $paramMetadatas = @()
     $paramMetadatas += $repairedMetadata.Parameters.Values
 
-    $conflictingParams = Get-ConflictingParameterNames
+    $conflictingParams = $script:ConflictingParameterNames
```

Comment:
As you mentioned in your last comment, using the `Get-ConflictingParameterNames` functions allows us to mock it during our tests executions.

Please revert this change

Reply:
Imho this is fine, you just need to set the script scope variable before running the test, instead of mocking the function. In fact you can just resolve the variable without qualifier here, which will allow you to set it in the above scope by using the `InModuleScope` block without overwriting data for the rest of the module. This also means that you don't have to remember the data that were in the variable and putting them back. 

example: 

```powershell
Get-Module m | Remove-Module

$module = New-Module -Name m -ScriptBlock { 
    $script:ConflictingParameterNames = "a", "b"

    function Get-Names {
        "without qualification"
        $ConflictingParameterNames
        "with qualification"
        $script:ConflictingParameterNames
    }
} 

$module | Import-Module

Get-Names
"---"
# this works like InModuleScope,
# it runs the code in the module scope
& ($module) {
    $ConflictingParameterNames = "f"
    Get-Names
}
"---"
Get-Names
```

Reply:
I wasn't able to get this to work, as far as I could tell tests for conflicting where already being run in the Pester module context (using InModuleScope) but setting the variable unqualified still didn't work, so I've left the Get command in to return the variable to make testing easier.